agent/lib/kontena/workers/stats_worker.rb

```diff
@@ -30,6 +30,9 @@ def configure_statsd(node)
       else
         @statsd = nil
       end
+    rescue => error
+      warn "statsd configuration failed: #{error.message}"
+      warn error.backtrace.join('\n') if error.backtrace
```

Comment:
Something like that would also be needed to fix #1653, but it's not really relevant for this change to workaround #2231